src/Common/tests/System/Net/VirtualNetwork/VirtualNetwork.cs

```diff
@@ -36,7 +36,24 @@ public void ReadFrame(bool server, out byte[] buffer)
 
             semaphore.Wait(WaitForReadDataTimeoutMilliseconds);
 
-            bool dequeueSucceeded = packetQueue.TryDequeue(out buffer);
+            bool dequeueSucceeded = false;
+            int remainingRetries = 3;
+            int backOffDelayMilliseconds = 10;
+
+            do
+            {
+                dequeueSucceeded = packetQueue.TryDequeue(out buffer);
+                if (dequeueSucceeded)
+                {
+                    break;
+                }
+
+                remainingRetries--;
+                backOffDelayMilliseconds *= backOffDelayMilliseconds;
+                Thread.Sleep(backOffDelayMilliseconds);
```

Comment:
This math doesn't seem quite right.  After the first try, we're going to sleep for 10*10 == 100 milliseconds, that's fine.  The second time we're going to sleep for 100*100 == 10,000 millseconds = 10 seconds... a bit long, but maybe ok.  The third time we're going to sleep for 10000*10000 milliseconds == 27 hours?

Did you mean to be multiplying by 10 each time rather than by backoffDelayMilliseconds?